### ISSUE TITLE
Support Scripted Metric Aggregation

### ISSUE THREAD

**Issue by zanes2016** [olivere/elastic#1013]
Is there currently support for [Scripted Metric Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-metrics-scripted-metric-aggregation.html)?

### Elastic Version

- elastic.v6 (for Elasticsearch 6.3)

### Description
I may just be missing the function in the GoDoc, but is there support for Scripted Metric Aggregations? I don't see any `new ScriptedMetricAggregation()` function like there are for the other aggregation types (i.e. SumAggregation()).

**Comment by olivere** [comment]
You're right, it's not implemented yet. Will have a look at it for the next release.

Note to future self: [`ScriptedMetricAggregationBuilder.java`](https://github.com/elastic/elasticsearch/blob/6.x/server/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricAggregationBuilder.java)

**Comment by larrycinnabar** [comment]
@olivere PR is ready for this issue. I used this piece of code on production with real scripted metric aggregations (different ones) for a month already

**Comment by olivere** [comment]
@larrycinnabar Yep, it seems complete. I currently plan to put it into 6.2.16, due (hopefully) next weekend. Thanks for your support!
